app/controllers/user_reports_controller.rb

```diff
@@ -5,10 +5,11 @@ class UserReportsController < ApplicationController
   before_action :load_course, except: %i(show destroy)
 
   def index
-    @user_reports = current_user.user_reports
-                                .includes(:course).search(params)
-                                .recent.page(params[:page])
-                                .per Settings.course_index_page
+    @q = current_user.user_reports.ransack(params[:q])
+    @q.sorts = UserReport::SORT_PARAMS if @q.sorts.empty?
```

Comment:
```suggestion
    @q.sorts = params.dig(:q, :s) if params.dig(:q, :s)
```
Thử cách này xem em, xem còn cần sort_params không

Reply:
Em nghĩ anh hiểu nhầm đoạn này ạ.

Ở đây em đưa SORT_PARAMS vào là để sắp xếp mặc định khi không có tham số sắp xếp nào được truyền vào ạ, còn nếu đã có tham số sắp xếp truyền vào thì vẫn giữ nguyên và sắp xếp theo tham số truyền vào đó.

Có nghĩa là câu lệnh của anh gợi ý vốn nó đã là câu lệnh ẩn rồi ạ. Mặc định là @q.sorts = params.dig(:q, :s) dù cho :s có tồn tại hay không.

Nên vẫn cần SORT_PARAMS ạ.